x/outline-connectivity-app/shared_backend/index.ts

```diff
@@ -0,0 +1,25 @@
+import { ConnectivityTestInput, ConnectivityTestResult } from "./generated/types"
+
+export function from(backend: Invokable) {
+  return {
+    connectivityTest: invocationFactory<ConnectivityTestInput, ConnectivityTestResult[]>(backend, "ConnectivityTest")
+  }
+}
+
+/* INFRASTRUCTURE (to be moved) */
+
+export interface Invokable {
+  Invoke(parameters: { method: string; input: string; }): Promise<{ result: string; errors: string[] }>
+}
+
+function invocationFactory<T, K>({ Invoke }: Invokable, methodName: string) {
+  return async (input: T): Promise<K> => {
+    const { result, errors } = await Invoke({ method: methodName, input: JSON.stringify(input) });
```

Comment:
If we inline the parameters object, the call will be more readable:
```suggestion
    const { result, errors } = await Invoke(methodName, JSON.stringify(input));
```